src/Castle.Windsor.Extensions.DependencyInjection/Scope/ExtensionContainerScope.cs

```diff
@@ -69,15 +69,14 @@ public Burden GetCachedInstance(ComponentModel model, ScopedInstanceActivationCa
 				scopeCache[burden] = burden;
```

Comment:
You are locking access to `scopeCache` below, but what about this code that is nearly a duplicate. Maybe refactor it?

Reply:
> You are locking access to `scopeCache` below, but what about this code that is nearly a duplicate. Maybe refactor it?

I will look. Yes there are issues with the original code that still needs to be cleaned up.

Reply:
@jonorossi please look at my change. If I need to refactor I will have to lock it all:

It appears that this is how the scopes are being managed by the facility. It isn't using Castle Scoping directly, I think this was a sort of wrapper.
Is this the correct way to go? and thoughts @jonorossi @AntonioDrusin 


```c#
public Burden GetCachedInstance(ComponentModel model, ScopedInstanceActivationCallback createInstance)
		{
			lock (scopeCache)
			{
				Burden burden = null;
				if(model.Configuration.Attributes.Get(TransientMarker) != bool.TrueString )
				{
					burden = scopeCache[model];
				}
				if (burden != null) return burden;
				scopeCache[model] = burden = createInstance((_) => {});
				return burden;
			}
		}
```